### PR TITLE
Issue #211 S1172 should not report on event handlers

### LINKED ISSUE #211: S1172 should not report on event handlers
S1172 should not flag unused members if they are part of an interface method or when they are  passed to a delegate/eventhandler.

Classic example of this is the (sender, event) passed to an Event handling function. Many people don't use the sender and often not even the event, this is not really a problem.

Alternative syntax, such as += (s,e) => MethodWithoutParameters(); works, but had the problem that unsubscribing from the event requires us to store the delegate so that we can use it in the -= call later on.

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #211

### CODE REVIEW

File: src/SonarLint/Rules/MethodParameterUnused.cs [hunk 1]
```diff
@@ -124,5 +125,29 @@ private static bool IsMethodCandidate(IMethodSymbol methodSymbol)
...
+                methodSymbol.Parameters.Length != 2)
+            {
+                return false;
+            }
+
+            var eventArgs = methodSymbol.Parameters[1];
+            var eventArgsType = eventArgs.Type as INamedTypeSymbol;
+            if (eventArgsType == null)
+            {
+                return false;
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/MethodParameterUnused.cs]: @tvsonar the conservative option (prefer false positives over false negatives) here would be to return `true` if we don't know exactly what the type is

File: src/SonarLint/Rules/MethodParameterUnused.cs [hunk 2]
```diff
@@ -124,5 +125,29 @@ private static bool IsMethodCandidate(IMethodSymbol methodSymbol)
...
+            }
+
+            var sysEventArgs = compilation.GetTypeByMetadataName("System.EventArgs");
+            if (sysEventArgs == null)
+            {
+                return false;
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/MethodParameterUnused.cs]: @tvsonar the conservative option (prefer false positives over false negatives) here would be to return `true` if we don't know exactly what the type is

**Comment by dbolkensteyn** [comment]
LGTM

File: src/SonarLint/Rules/MethodParameterUnused.cs (hunk 2 above)
Comment (tvsonar) [inline src/SonarLint/Rules/MethodParameterUnused.cs]: Thanks, changed both of them.

**Pull request merged**